packages/x-charts/src/internals/store/useCharts.ts

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import { ChartStore } from '../plugins/utils/ChartStore';
+import {
+  ChartAnyPluginSignature,
+  ChartInstance,
+  ChartPlugin,
+  ChartPublicAPI,
+  ChartState,
+  ConvertSignaturesIntoPlugins,
+} from '../plugins/models';
+import { CHART_CORE_PLUGINS, ChartCorePluginSignatures } from '../plugins/corePlugins';
+import { UseChartBaseProps } from './useCharts.types';
+import { UseChartInteractionState } from '../plugins/featurePlugins/useChartInteraction/useChartInteraction.types';
+
+export function useChartApiInitialization<T>(
+  inputApiRef: React.MutableRefObject<T | undefined> | undefined,
+): T {
+  const fallbackPublicApiRef = React.useRef({}) as React.MutableRefObject<T>;
+
+  if (inputApiRef) {
+    if (inputApiRef.current == null) {
+      // eslint-disable-next-line react-compiler/react-compiler
+      inputApiRef.current = {} as T;
+    }
+    return inputApiRef.current;
+  }
+
+  return fallbackPublicApiRef.current;
+}
+
+let globalId = 0;
+
+export function useCharts<
+  TSignatures extends readonly ChartAnyPluginSignature[],
+  TProps extends Partial<UseChartBaseProps<TSignatures>>,
+>(inPlugins: ConvertSignaturesIntoPlugins<TSignatures>, props: TProps) {
+  type TSignaturesWithCorePluginSignatures = readonly [
+    ...ChartCorePluginSignatures,
+    ...TSignatures,
+  ];
+
+  const plugins = React.useMemo(
+    () =>
+      [
+        ...CHART_CORE_PLUGINS,
+        ...inPlugins,
+      ] as unknown as ConvertSignaturesIntoPlugins<TSignaturesWithCorePluginSignatures>,
+    [inPlugins],
+  );
+
+  const pluginParams = {}; // To generate when plugins use params.
+  const instanceRef = React.useRef({} as ChartInstance<TSignatures>);
+  const instance = instanceRef.current as ChartInstance<TSignatures>;
+  const publicAPI = useChartApiInitialization<ChartPublicAPI<TSignatures>>(props.apiRef);
+  const innerSvgRef: React.RefObject<SVGSVGElement> = React.useRef(null);
+
+  const storeRef = React.useRef<ChartStore<TSignaturesWithCorePluginSignatures> | null>(null);
+  if (storeRef.current == null) {
+    // eslint-disable-next-line react-compiler/react-compiler
+    globalId += 1;
+
+    const initialState = {
+      // TODO remove when the interaction moves to plugin
```

Comment:
Is this TODO still relevant? 🤔

Reply:
Yes, because the interaction pluggin is created but not used.

To move interaction behavior I would need to migrate the `useAxisListener`, which implies to have access to the `drawingArea`

Reply:
Ah, got it